Cccp set

VinInn opened the pull request:
Factorize CCC value into a single cfi file
Factories the reading of this PSet into a single header file
use consistently.

No regression expected, no regression observed. HLT expected to fail.
@mtosi @perrotta please check/act/instruct

affected producers

SiStripClusterizer, SiStripClusterizerFromRaw (actually their  Algorithm''s PSet)
SeedingLayersEDProducer (actually its embedded strip PSets)
Chi2ChargeMeasurementEstimatorESProducer
ClusterChargeMasker
ClusterShapeHitFilterESProducer

in all cases the optional attribute
"minGoodStripCharge"/"minGoodCharge"
has been substituted with a MANDATORY

```
clusterChargeCut = cms.PSet(refToPSet_ = cms.string('SiStripClusterChargeCutXXX'))
```

with XXX = None/Loose/Tight
corresponding to current CCC values of 0/1724/2069

Lesson Learned:
use of ParameterSet::existsAs is nothing less that introducing a time bomb that will at some point induce subtle regressions such as massive HLT/ALCA silent failures...

rappoccio committed:
Adding first prototype at a 'jet pieces' interface for substructure

rappoccio committed:
Changing FwdPtr to Ptr as per Andrea's request. Also putting together the start of a workflow.

rappoccio committed:
Adding a version for comments, currently not working

rappoccio committed:
Disabling subjet flavor as per Dinko's suggestion

rappoccio committed:
First compiling and running version of packed substructure jets

rappoccio committed:
Fixing the Ptr constructor

rappoccio committed:
Adding a print method to pat::Jet and adjusting event content

rappoccio committed:
Refactorizing the python for substructure to a function to clean up the top level

rappoccio committed:
Fixing checksum

rappoccio committed:
Changing matched gen particles to be conssitent with others

rappoccio committed:
Changing distMin to distMax for Dinko

rappoccio committed:
Changing distMin to distMax in header

rappoccio committed:
Committed wrong file

VinInn committed:
move CCC in pset

VinInn committed:
move CCC in pset, first step done

VinInn committed:
migrate hitextractor as well

VinInn committed:
migrate chi2cut as well

VinInn committed:
fix default seed layers

VinInn committed:
fix special and hi

rappoccio committed:
Slimming subjets

VinInn committed:
remove cout

rappoccio committed:
Add the slimmed subjets

VinInn committed:
fix calZeroBiasClusters

VinInn committed:
fix Famos

VinInn committed:
fix import

cmsbuild commented:
A new Pull Request was created by @VinInn (Vincenzo Innocente) for CMSSW_7_4_X.

Cccp set

It involves the following packages:

Calibration/TkAlCaRecoProducers
FastSimulation/Tracking
RecoHI/HiTracking
RecoLocalTracker/SiStripClusterizer
RecoLocalTracker/SubCollectionProducers
RecoPixelVertexing/PixelLowPtUtilities
RecoTracker/ConversionSeedGenerators
RecoTracker/IterativeTracking
RecoTracker/MeasurementDet
RecoTracker/SpecialSeedGenerators
RecoTracker/TkSeedingLayers
RecoTracker/TrackProducer


@civanch, @diguida, @lveldere, @cvuosalo, @mdhildreth, @cmsbuild, @nclopezo, @cerminar, @slava77, @mmusich can you please review it and eventually sign? Thanks.
@ghellwig, @forthommel, @yduhm, @Martin-Grunewald, @threus, @venturia, @makortel, @richard-cms, @jlagram, @cerati, @mandrenguyen, @yetkinyilmaz, @GiacomoSguazzoni, @rovere, @VinInn, @echapon, @dgulhan, @nickmccoll, @gbenelli, @jazzitup, @appeltel, @yenjie, @kurtejung, @gpetruc, @matt-komm this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

Dr15Jones commented:
I concur with your "lessons learned".

lveldere commented on FastSimulation/Tracking/python/IterativePixelLessStep_cff.py:
```diff
@@ -33,6 +33,8 @@
 #                                     'TID1_neg+TID2_neg','TID2_neg+TID3_neg',
 #                                     'TEC1_neg+TEC2_neg','TEC2_neg+TEC3_neg','TEC3_neg+TEC4_neg','TEC3_neg+TEC5_neg','TEC4_neg+TEC5_neg']
+
+from RecoLocalTracker.SiStripClusterizer.SiStripClusterChargeCut_cfi import *
```
@VinInn 
How does this affect FastSim?
FastSim has no real charge.
Lukas

VinInn replied:
in the following line you import pixelLessStepSeedLayers that makes use of it
There is most probably a different way to import it silently, at the moment my expertise in python (and understanding of RefToPset) makes the current solution the easier

lveldere replied:
Are you sure this is needed"
We only use the layer list from the seedlayer, 
and copy it to the cfg of the fast trajectory seed producer.
The module itself is not run.

May be there is something I don't understand about the python processing...

VinInn replied:
It fails w/o
I acknowledge that there is definitively something "I" don't understand about the python processing of RefToPSet

makortel replied:
How does it fail? (I would also have expected this configuration to work without the import above)

VinInn replied:
```----- Begin Fatal Exception 11-Feb-2015 12:40:37 CET-----------------------
An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named ZEE_13TeV_TuneCUETP8M1_cfi_GEN_SIM_RECO_EI_HLT_VALIDATION.py
Exception Message:
python encountered the error: <type 'exceptions.AttributeError'>
'Process' object has no attribute 'SiStripClusterChargeCutTight'
```

makortel replied:
I think what happens is the following
- from the import below it follows that a `process` loading this configuration will have `process.pixelLessStepSeedLayers` member, that has `refToPSet_`s to `SiStripClusterChargeCutTight`
- all `refToPSet_`s are resolved when the `process` is "transformed" to C++ ParameterSets (`cms.Process.fillProcessDesc()`)
- if the import above is missing, the `process` does not have the `SiStripClusterChargeCutTight` member referenced by a `refToPSet_` in `process.pixelLessStepSeedLayers`

I have one idea to circumvent this "behind the scenes" dependence, but I want to test it first myself.

Martin-Grunewald commented:
Hi @VinInn 
The statement ```HLT expected to fail.``` is kind of useless!
Please do provide instructions on what do change in HLT!

venturia commented:
I am fine to couple  calZeroBiasClusters to the standard siStripClusters module but someone has to inform strip local reco people and be sure that this information is written in the documentation of the bad channel workflow. Next DPG meeting?

VinInn commented:
@venturia, I find cleaner, clearer and more maintainable to clone and modify the few (in this case one) parameters that differs w/r/t default than copy/paste/edit

venturia commented:
@VinInn . Agreed, but they have to know that now they depend on something they did not depend on before.

makortel committed:
Remove the import of SiStripClusterChargeCut_cfi from FastSim tracking configurations

The import of SiStripClusterChargeCut_cfi was needed because the
import of seeding layer configuration adds the *SeedLayers* modules to
the cms.Process when load()ed. If the *SeedLayers* modules are
imported with "as" and a name beginning with an underscore, these
names do not end up in cms.Process, and we can avoid importing
SiStripClusterChargeCut_cfi here.

cmsbuild commented:
Pull request #7673 was updated. @civanch, @diguida, @lveldere, @cvuosalo, @mdhildreth, @cmsbuild, @nclopezo, @cerminar, @slava77, @mmusich can you please check and sign again.

ferencek committed:
applied additional slimming of substructure info

cvuosalo commented:
@cmsbuild please test

slava77 commented:
it's somewhat pointless to test now, workflows including HLT will fail.
The changes needed for HLT should be provided.

cmsbuild commented:
The tests are being triggered in jenkins.

Martin-Grunewald commented:
@slava77 
This is not planned: I do not know which release this will end up with, 74X or 75X.
I already wasted time on https://github.com/cms-sw/cmssw/pull/7589 which
David Lange told me will after all not go in 74X but rather 75X.
So I do not see the point making any HLT changes unless we know for a fact
that it goes in 74X, and also when!

slava77 commented:
@Martin-Grunewald 
I think this is for 75X.

davidlange6 commented:
But indeed, needing to know the release that a new development goes into is imho the wrong approach. If you need to hack around, at least hack around in the backport as then its clear when the hack can be removed.

cmsbuild commented:
-1 
Tested at: 809b29e255eba77708bbb0cdf4a7a0d02c3c9f67
 When I ran the RelVals I found an error in the following worklfows: 
 135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

----- Begin Fatal Exception 12-Feb-2015 20:30:22 CET-----------------------

An exception of category 'Configuration' occurred while

   [0] Constructing the EventProcessor

   [1] Constructing ESProducer: class=Chi2ChargeMeasurementEstimatorESProducer label='hltESPChi2ChargeMeasurementEstimator16'

Exception Message:

MissingParameter: ParameterSet 'clusterChargeCut' not found.

----- End Fatal Exception -------------------------------------------------

</pre>
1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 12-Feb-2015 20:31:59 CET-----------------------

An exception of category 'Configuration' occurred while

   [0] Constructing the EventProcessor

   [1] Constructing ESProducer: class=Chi2ChargeMeasurementEstimatorESProducer label='hltESPChi2ChargeMeasurementEstimator16'

Exception Message:

MissingParameter: ParameterSet 'clusterChargeCut' not found.

----- End Fatal Exception -------------------------------------------------

</pre>
1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 12-Feb-2015 20:36:10 CET-----------------------

An exception of category 'Configuration' occurred while

   [0] Constructing the EventProcessor

   [1] Constructing ESProducer: class=Chi2ChargeMeasurementEstimatorESProducer label='hltESPChi2ChargeMeasurementEstimator16'

Exception Message:

MissingParameter: ParameterSet 'clusterChargeCut' not found.

----- End Fatal Exception -------------------------------------------------

</pre>
25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15+MINIAODMCUP15/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15+MINIAODMCUP15.log

----- Begin Fatal Exception 12-Feb-2015 20:41:36 CET-----------------------

An exception of category 'Configuration' occurred while

   [0] Constructing the EventProcessor

   [1] Constructing ESProducer: class=Chi2ChargeMeasurementEstimatorESProducer label='hltESPChi2ChargeMeasurementEstimator16'

Exception Message:

MissingParameter: ParameterSet 'clusterChargeCut' not found.

----- End Fatal Exception -------------------------------------------------

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7673/2543/summary.html

Martin-Grunewald commented:
@davidlange6 
David, you are so long with CMSSW that you should know the constraints
TSG with ConfDB is operating under so refrain from such comments.
If you think TSG does crap have a chat with Andrea B.

davidlange6 commented:
I didn't say anything to suggest this.. rather a better way to live within the constraints imposed by the TSG.

On Feb 13, 2015, at 7:01 AM, Martin Grunewald <notifications@github.com> wrote:

> @davidlange6 
> David, you are so long with CMSSW that you should know the constraints
> TSG with ConfDB is operating under so refrain from such comments.
> If you think TSG does crap have a chat with Andrea B.
> 
> �
> Reply to this email directly or view it on GitHub.
>

davidlange6 commented:
closing in 74x - there is the same PR open in 75x (now the development release)

jshlee committed:
adding puppi to miniAODs
added puppi jets and met
added puppi weights to packcandidates

jshlee committed:
removed bool for checking if collection exists

jshlee committed:
added jet charge, btagging, secondary vertex info for puppi jets

jshlee committed:
changed packed puppi weight
to pack8log( (p-0.5)*2,-2,0,64)
liblogintpack bug fix - negative not saved for r=0 or r=base-1

jshlee committed:
changed btagDiscriminators mapped to
process.patJets.discriminatorSources

gpetruc committed:
AK8 constituents not in subjets added as daughters in addition
to the subjets, so that alternative reclusterings are possible

jshlee committed:
trying to add puppiJet Daughters

VinInn committed:
increase tollerance to make space for shifted FPIX

jshlee committed:
Update PuppiProducer.cc

arizzi committed:
fixing the valumaps

venturia committed:
New macro and script to extract occupancy plots from ALCARECO DQM files

jshlee committed:
adding PuppiSrc and PuppiSrcMap to packedPFCandidates_cfi

jshlee committed:
removing obsolete part that was removed in #8132
but added back during merging

jshlee committed:
removing the label in puppi producer

Commit:
Update the heavy ion trigger menu XML file to use the latest

jmmans committed:
Fix compiler warning for QIEXXDataFrames
Replace "magic numbers" with constants
Add "flag" word which will inevitably be required operationally

Commit:
Optimize SiStripMonitorTrack to perform less string operations.

https://hypernews.cern.ch/HyperNews/CMS/get/swDevelopment/3171.html

wmtan committed:
ROOT6 checksums must be known to ROOT5

wmtan committed:
Remove unnecessary ROOT5/ROOT6 difference in SimDataFormats/PileupSummaryInfo

wmtan committed:
Remove unnecessary ROOT5/ROOT6 diffs in CondCore/IOVService

Commit:
enforce no DS naming collisions (relevant only for submission to production)

Commit:
miniaod validation in full, with unscheduled execution of PAT+RECO in the same step

wmtan committed:
Drop unneeded explicit dependency

VinInn committed:
Merged CMSSW_7_4_0_pre8/sign527/pull8174-cp from repository slava77

venturia committed:
Added the possibility to produce a root file with the selected occupancy plots

venturia committed:
Create README.md

gpetruc committed:
Merged refs/pull/8249/head from repository cms-sw

ggovi committed:
Default format for new condition databases moved to v2

gpetruc committed:
Better packing of puppi weights, preserving zero and one

gpetruc committed:
Fix the linking of puppi candidates after sorting (and also of tracks, since it had been broken by mistake in #8133)

jshlee committed:
Merge pull request #296 from gpetruc/fixPuppi_74X

Fix daughter references, and better packing of puppi weights.

arizzi committed:
do not ignore the input puppi value map

jshlee committed:
Merge pull request #297 from arizzi/puppi

do not ignore the input puppi value map

Commit:
New binning for the jet mutliplicity histogram

ferencek committed:
removed 'using namespace' statements and updated the code accordingly

wmtan committed:
Remove unnecessary direct dependencies on rootrflx

wmtan committed:
Backport changes for ROOT6 that are OK for ROOT5

wmtan committed:
Add missing std::

Dr15Jones committed:
Calling TFormula::SetName leads to thread safety issues

When a TFormula is created, it may be added to a global list of formulas.
If that happens, then calling TFormula::SetName at the same time another
thread is looping through the list of formulas can lead to a crash.
Setting the name correctly in the constructor avoids the problem.
This problem was found by helgrind.

wmtan committed:
Back port changes from ROOT6 IB

ktf committed:
Merge pull request #8293 from wmtan/ReconcileFireworks

Add missing std::

ktf committed:
Merge pull request #8283 from wmtan/ReconcileIOPoolStreamer

Backport IOPool/Streamer changes for ROOT6 that are OK for ROOT5

cmsbuild committed:
Merge pull request #8233 from wmtan/ReconcileSimDataFormatsPileupSummaryInfo

Remove unnecessary ROOT5/ROOT6 difference in SimDataFormats/PileupSummar...

cmsbuild committed:
Merge pull request #8247 from VinInn/FixFPIXfor74X

Workaround for FPIX shift for 74X

davidlange6 committed:
Merge pull request #8267 from ebouvier/JetMultBinningTopDQMOffline74X

New binning for the jet multiplicity histogram in Top DMQ Offline

davidlange6 committed:
Merge pull request #8245 from wmtan/DropUnneededDependencies

Drop unneeded explicit dependency in 7_4_X

davidlange6 committed:
Merge pull request #8292 from wmtan/ReconcileAnalysis

Back port analysis changes from ROOT6 IB

davidlange6 committed:
Merge pull request #8281 from wmtan/RemoveUnneededrootrflx

Remove two remaining unnecessary direct dependencies on rootrflx

davidlange6 committed:
Merge pull request #8239 from wmtan/ReconcileCondCoreIOVService

Remove unnecessary ROOT5/ROOT6 diffs in CondCore/IOVService

davidlange6 committed:
Merge pull request #8207 from UMN-CMS/7_4_qie_patch

Fix compiler warning for QIEXXDataFrames (Backport of #8206)

davidlange6 committed:
Merge pull request #8228 from wmtan/ROOT6ChecksumsInROOT5

ROOT6 checksums must be known to ROOT5

cmsbuild committed:
Merge pull request #8242 from franzoni/miniAodValidation-CMSSW_7_4_X_2015-03-12-1400

MiniAod validation and DQM

davidlange6 committed:
Merge pull request #8205 from mark-grimes/updateEras

Update the Run2_HI era to use the latest trigger menu XML file

cmsbuild committed:
Merge pull request #8300 from threus/script_fix_74x

TK offline shift tools fixes

cmsbuild committed:
Merge pull request #8298 from venturia/sistripmonitorclient_74x

New script and macro to extract occupancy plots from DQM files

cmsbuild committed:
Merge pull request #8289 from Dr15Jones/fixRecoTauDiscriminantCutMultiplexerThreadSafety

Calling TFormula::SetName leads to thread safety issues

cmsbuild committed:
Merge pull request #8278 from cms-btv-pog/IPProducerUpdate_from-CMSSW_7_4_0_pre7

IPProducer update: Part 2

cmsbuild committed:
Merge pull request #8266 from dmitrijus/wednesday_fix_for_fede

Optimize SiStripMonitorTrack to perform less string operations.

cmsbuild committed:
Merge pull request #8249 from jshlee/puppiIntoMiniAOD

Puppi into mini aod

cmsbuild committed:
Merge pull request #8175 from rappoccio/MiniAODSubstructurePiecesAugmented

Mini aod substructure pieces overhaul, rebase of #7659

cmsbuild committed:
Merge pull request #8259 from ggovi/new-conddb-sqlite-format-v2-74X

Default format for new condition databases moved to v2

Martin-Grunewald commented:
@VinInn 

Please re-open this PR.

VinInn commented:
was closed by @davidlange6

Martin-Grunewald commented:
@VinInn 
While TTRH does not have merge conflicts, this one now has - 
could you please have a look fix?

Martin-Grunewald commented:
It seems the only merge conflict is in 
```
 RecoLocalTracker/SiStripClusterizer/python/DefaultClusterizer_cff.py
```
where some other parameters have been removed in the meantime.
Could you please update this PR for that (and its 75X twin)?
Thanks

VinInn committed:
merge with Bill F. changes

VinInn commented:
should be ok now

cmsbuild commented:
Pull request #7673 was updated. @diguida, @lveldere, @cerminar, @Martin-Grunewald, @perrotta, @civanch, @srimanob, @cmsbuild, @fwyzard, @ktf, @Dr15Jones, @cvuosalo, @mdhildreth, @deguio, @slava77, @ggovi, @vadler, @mmusich, @mulhearn, @ssekmen, @nclopezo, @danduggan, @boudoul, @monttj, @alja, @franzoni can you please check and sign again.

slava77 commented:
-1

superseded by #8343
